website/source/docs/providers/aws/r/db_instance.html.markdown

```diff
@@ -34,9 +34,9 @@ the [AWS official documentation](http://docs.aws.amazon.com/AmazonRDS/latest/Com
 
 The following arguments are supported:
 
-* `allocated_storage` - (Required) The allocated storage in gigabytes.
-* `engine` - (Required) The database engine to use.
-* `engine_version` - (Required) The engine version to use.
+* `allocated_storage` - (Optional when a `snapshot_identifier` is provided) The allocated storage in gigabytes.
```

Comment:
This is totally a nitpick, but I think I'd phrase this as "Required unless a `snapshot_identifier` is provided" to describe the situation with slightly more precision.